.github/workflows/conda.yml

```diff
@@ -5,6 +5,12 @@ on:
       - main
       - datafusion-sql-planner
   pull_request:
+    paths:
+      - setup.py
+      - continuous_integration/recipe/**
+      - .github/workflows/conda.yml
+  schedule:
+    - cron: '0 0 * * 0'
```

Comment:
Since these changes make conda builds take up much more time, this makes it so they only run when relevant files are impacted in PRs (and weekly to make sure things are working as expected)